pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -22,33 +22,26 @@ type ServiceBindingRequestSpec struct {
 	// +optional
 	CustomEnvVar []corev1.EnvVar `json:"customEnvVar,omitempty"`
 
-	// BackingServiceSelector is used to identify the backing service operator.
-	// Deprecation Notice:
-	// In the upcoming release, this field would be depcreated. It would be mandatory
-	// to set "backingServiceSelectors".
-	// +optional
-	BackingServiceSelector *BackingServiceSelector `json:"backingServiceSelector,omitempty"`
-
-	// BackingServiceSelectors is used to identify multiple backing services.
-	// This would be made a required field after 'BackingServiceSelector'
+	// Services is used to identify multiple backing services.
+	// This would be made a required field after 'Services'
 	// is removed.
 	// +optional
-	BackingServiceSelectors *[]BackingServiceSelector `json:"backingServiceSelectors,omitempty"`
+	Services *[]Services `json:"services,omitempty"`
 
-	// ApplicationSelector is used to identify the application connecting to the
+	// Application is used to identify the application connecting to the
 	// backing service operator.
 	// +optional
-	ApplicationSelector ApplicationSelector `json:"applicationSelector"`
+	Application *Application `json:"application"`
 
 	// DetectBindingResources is flag used to bind all non-bindable variables from
 	// different subresources owned by backing operator CR.
 	// +optional
-	DetectBindingResources bool `json:"detectBindingResources,omitempty"`
+	DetectBindingResources *bool `json:"detectBindingResources,omitempty"`
```

Comment:
@pedjak I was referring changes made on https://github.com/redhat-developer/service-binding-operator/pull/495, @sbose78 has mentioned this are the changes are the result of feedback he's received from OLM team. Anything to add to it @sbose78 ?

Reply:
Helps us differentiate between false and unspecified :)

Reply:
any particular usecase in mind? When unspecified, we should assume some default value, and that is implicit to user, and that could be confusing. If we do not have any concrete usecase, I would avoid this for now.

Reply:
I would do it to [avoid appearance of empty fields](https://github.com/redhat-developer/service-binding-operator/pull/369) and to be consistent with other changes in the current PR.

In the current PR, we are doing the same for `Application` too and hence feels natural to me that we would do the same for other attributes as well.

I would rather argue, 
* Do we have a concrete use case to show users fields they never cared about to begin with? If not, we should do the needful to avoid them showing up. 
* I'm OK with the opposite as well - as long as we are consistent.


Here was the feedback from the OLM team:

```
Hello,

Did a quick pass through just some things to consider:

The Application is optional but I don’t think that I would be able to set nil here. Making it a pointer to an Application I think is what we should do.

The DetectBindingResources is optional, but has a default of false. Is there ever going to potentially be a time when a non-set value is going to be different then a false value?

The Application type has a label selector and a LocalObjectReference do I need to set a name and then use a label selector? I think that these should be a union type

Thanks,

Shawn Hurley
```

Reply:
@sbose78 so the decision depends on the answer to "The DetectBindingResources is optional, but has a default of false. Is there ever going to potentially be a time when a non-set value is going to be different then a false value?" 

Currently, our code assumes that the value is false, if omitted. IMHO, I find it less surprising if the default value is `false`, it is something users are used to in all mainstream languages.

https://stackoverflow.com/questions/14157180/why-does-the-default-value-of-a-boolean-variable-tend-to-be-false

Reply:
> so the decision depends on the answer

Not necessarily, as in, not how I reasoned this change.

I proposed it to be a pointer for two reasons across all spec attributes.

* So that everything optional is nil-able.
* If the user doesn't set it, it shouldn't show up post creation.

No strong opinion here 🤷‍♂️  We could go either way.

Reply:
> We could go either way.

IThanks for providing the details about the reasoning. Still not sure if this particular field should be declared as optional, but instead of splitting hair further, let's see how this is going to work in future. We can always change that later.